### PR TITLE
Pull out peercoin logic from btcpy code.

### PR DESCRIPTION
**Pull request by belovachap**
@peerchemist @saeveritt 

This pull request moves the Peercoin specific logic into subclasses. The unit test suite is passing again (`python -m unittest tests.unit`)!

### CODE REVIEW

File: btcpy/structs/transaction.py [hunk 1]
```diff
@@ -188,15 +188,19 @@ class TxOut(Immutable, HexSerializable, Jsonizable):
...
     @classmethod
-    def from_json(cls, dic):
-        return cls(int(Decimal(dic['value']) * Constants.get('to_unit')),
-                   dic['n'],
-                   ScriptBuilder.identify(bytearray(unhexlify(dic['scriptPubKey']['hex']))))
-
-    def __init__(self, value: int, n: int, script_pubkey: ScriptPubKey):
+    def from_json(cls, dic, network):
+        return cls(
+            int(Decimal(dic['value']) * network.decimals),
```
Comment (peerchemist) [inline btcpy/structs/transaction.py]: Please check this commit: https://github.com/PeerAssets/btcpy/commit/73f22bf998fc5bbafa29d28045a36acbb75cc710

You need to move from "network.decimals" to appropriate new entry in Constants.

Comment (belovachap) [reply btcpy/structs/transaction.py]: :+1: will patch up

**Comment by belovachap** [comment]
I added a lazy test to convert the test transactions to json and back again... all of them pass except the last one :\ might also be an upstream issue?

```
FAIL: test_json (tests.unit.TestTransaction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/chapman/code/btcpy/tests/unit.py", line 272, in test_json
    self.assertEqual(tx, tx_from_json)
AssertionError: <btcpy.structs.transaction.SegWitTransaction object at 0x7f53f60a4b00> != <btcpy.structs.transaction.Transaction object at 0x7f53f609ae80>
```

**Comment by peerchemist** [comment]
Could be, have you tested the patch against upstream? If it is their problem report it.

**Comment by belovachap** [comment]
Yeah, I don't have a fix for it yet but I think it doesn't reconstitute SegWit transaction from json at the moment https://github.com/chainside/btcpy/pull/42

**Pull request merged**